staging/src/k8s.io/legacy-cloud-providers/openstack/openstack_volumes.go

```diff
@@ -731,15 +731,15 @@ func (os *OpenStack) GetLabelsForVolume(ctx context.Context, pv *v1.PersistentVo
 		return nil, nil
 	}
 
-	// Get Volume
-	volume, err := os.getVolume(pv.Spec.Cinder.VolumeID)
+	// Get metadata
+	md, err := getMetadata(os.metadataOpts.SearchOrder)
```

Comment:
@Fedosin why is this a carry and not upstreamed?

Reply:
Given that @Fedosin appears to be on vacation, I think this concern can be deferred for now. I've prompted him on slack so hopefully he answers when he returns.

Reply:
Hi! I did this because there was a requirement to read cloud credentials from a secret. The upstream version doesn't support it, as you know.
As I remember, there was an issue with kube-apiserver to get a volume info, so I implemented this workaround.

I believe we can revert it now and go back to the upstream implementation.